### ISSUE TITLE
External enums not re-exported from generated `types.ts` break near operations

### ISSUE THREAD

**Issue by vhenzl** [dotansimha/graphql-code-generator#2206]
**Describe the bug**
If external enums (either explicit mapping `Role:./enums#MyRole` or the new single file option #2038) used in combination with `near-operation-file` preset and `typescript-operations`, and an enum is used as a field argument, the enum is incorrectly imported in the "near" file. 
Or in other words, external enums aren't re-exported from generated `types.ts`.

```ts
// src/UsersByRole.generated.ts
import * as Types from "./types";

export type UsersByRoleQueryVariables = {
  role: Types.Role; // 💥 has no exported member 'Role'
};

export type UsersByRoleQuery = { __typename?: "Query" } & {
  usersByRole: Array<
    { __typename?: "User" } & Pick<Types.User, "id" | "username">
  >;
};

```

**To Reproduce**

GraphQL schema:

```graphql
# schema.graphql
schema {
  query: Query
}

type Query {
  usersByRole(role: Role!): [User!]!
}

enum Role {
  USER,
  ADMIN,
}

type User {
  id: ID!
  username: String!
  role: Role!
}
```

GraphQL operations:

```graphql
# src/UsersByRole.graphql
query UsersByRole($role: Role!) {
  usersByRole(role: $role) {
    id
    username
  }
}
```

Config file:

```yml
# codegen.yml
schema: schema.graphql
documents: src/**/*.graphql
config:
  enumValues:
    Role: ./enums#MyRole
generates:
  src/types.ts:
    - typescript
  src/:
    preset: near-operation-file
    presetConfig:
      baseTypesPath: types.ts
    plugins:
      - typescript-operations
```

File with external enums:

```ts
// src/enums.ts
export enum MyRole {
    User = "USER",
    Admin = "ADMIN",
}
```

Code sandbox example: https://codesandbox.io/s/graphql-codegen-issue-template-g5m7g

**Expected behaviour**
Generated `UsersByRole.generated.ts` correctly imports `Role` enum. Either directly from `enums.ts` or, probably better, from `types.ts` where is re-exported from `enums.ts`. 

**Environment:**

- OS: Win 10
- `@graphql-codegen/...`: canary, 1.4.0, 1.3.1,…
- NodeJS: 10

**Comment by dotansimha** [comment]
Thank you for reporting this, @vhenzl ! I think adding a simple `export` for the enums should do the trick, what do you think?

**Comment by vhenzl** [comment]
In the case of explicit mapping, there can be more files and therefore more exports. But yeah, reexport everything from the file(s) should work. 

What I see as a drawback is that the files could possibly contain additional exports which would be also reexported. And potentially it could leads to conflicts in exports.

Since all the enums are imported one by one anyway, it might be safer to reexport them explicitly one by one as well. 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by dotansimha** [comment]
Fixed in 1.5.0 🎉
